.github/workflows/run-test.yaml

```diff
@@ -0,0 +1,168 @@
+---
+name: Run these tests on a VM
+on:
+  workflow_dispatch:
+    inputs:
+      baseline:
+        description: The baseline version of Fluent Bit to compare against.
+        required: false
+        type: string
+        default: fluent/fluent-bit:latest
+      ref:
+        description: The branch name to use for comparison against, this will be built as a container to compare with.
+        required: true
+        type: string
+        default: https://github.com/fluent/fluent-bit.git#master
+      custom-config-ref:
```

Comment:
My thinking here is we can either have a whole repo with a test case, or a repo with a set of test cases then the custom-environment can specify the correct context/compose stack.